lib/compat/wordpress-6.8/block-editor-settings.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Block Editor Settings compatibility for WordPress 6.8
+ *
+ * Note: This will probably be a part of the get_default_block_editor_settings function during backport.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Extends the block editor settings with image format support checks.
+ *
+ * @param array $settings The current block editor settings.
+ * @return array Modified settings.
+ */
+function gutenberg_extend_block_editor_settings_with_image_support( $settings ) {
+	$settings['typesNotSupportedByServer'] = array();
+
+	// Check if WebP images can be edited.
+	if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+		$settings['typesNotSupportedByServer']['image/webp'] = true;
+	}
+
+	// Check if AVIF images can be edited.
+	if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/avif' ) ) ) {
+		$settings['typesNotSupportedByServer']['image/avif'] = true;
+	}
+
+	// Check if HEIC images can be edited.
+	if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/heic' ) ) ) {
+		$settings['typesNotSupportedByServer']['image/heic'] = true;
+	}
```

Comment:
Instead of this hardcoded list, can we just go through all image mime types returned by e.g. `wp_get_mime_types()` and check for support?

Reply:
> That list of types is [very long](https://github.com/WordPress/wordpress-develop/blob/3ff586a4e38681389d51206245750e5b72887cf6/src/wp-includes/functions.php#L3444-L3554) and includes many non image types. And some types actually don't need server support to enable uploading. For example, a PDF can be uploaded regardless of mime types support (with support, you get nice thumbnails though).

Hence my suggestion to go only through _image_ mime types, e.g. all starting with `image/`

This would also nicely cover mime types like `image/heif`, or `image/heic-sequence`, which are currently missing in this PR.

Reply:
> Hence my suggestion to go only through image mime types, i.e. all starting with image/

Ah, that makes sense. I will give that a try.

Reply:
Added in https://github.com/WordPress/gutenberg/pull/68644/commits/ee096abc56bfc68bc3b3d6889b98e72b2d61a6c5